netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -255,7 +242,13 @@ private NettyClientHandler newHandler() {
     frameWriter = new Http2OutboundFrameLogger(frameWriter, frameLogger);
 
     BufferingHttp2ConnectionEncoder encoder = new BufferingHttp2ConnectionEncoder(
-            new DefaultHttp2ConnectionEncoder(connection, frameWriter));
+            new DefaultHttp2ConnectionEncoder(connection, frameWriter)) {
+      @Override
+      public ChannelFuture writeSettingsAck(ChannelHandlerContext ctx, ChannelPromise promise) {
+        listener.transportReady();
```

Comment:
I think this is called to write acks for a receiving SETTINGS frame, and a gRpc server only send one SETTINGS frame at the start of the connection, so we also only write one ack for it, right?